src/einsteinpy/coordinates/conversion.py

```diff
@@ -71,380 +78,120 @@ def values(self):
 
         return self.t_si, self.x_si, self.y_si, self.z_si
 
-    def convert_spherical(self, **kwargs):
+
+class CartesianConversion(CoordinateConversionBase):
+    """
+    Class for conversion to and from Cartesian Coordinates in SI units
+    """
+
+    def convert_spherical(self) -> Tuple[float, float, float, float]:
         """
         Converts to Spherical Polar Coordinates
 
-        Other Parameters
-        ----------------
-        **kwargs : dict
-            Keyword Arguments
-
         Returns
         -------
         tuple
-            4-Tuple or 7-Tuple, containing the components in \
-            Spherical Polar Coordinates
-
+            4-Tuple containing the components in Spherical Polar Coordinates
         """
-        return cartesian_to_spherical_fast(
-            self.t_si,
-            self.x_si,
-            self.y_si,
-            self.z_si,
-            self.v_x_si,
-            self.v_y_si,
-            self.v_z_si,
-            self._velocities_provided,
-        )
+        return cartesian_to_spherical_fast(*self.values())
```

Comment:
Given [L68](https://github.com/einsteinpy/einsteinpy/blob/531c844ae827fe6541fb93a1ec2804b3c5728272/src/einsteinpy/coordinates/conversion.py#L68), this unpacking won't work, since all of these utility functions expect a `_velocities_provided` boolean parameter. A good solution here would be to refactor these functions in `coordinates/utils.py` to separate out the position and velocity conversion parts and to reassemble them, based on whether the velocity components (`u0, ..., u3`) are `None`. Something like so:
```py
# utils.py
def bl_to_cartesian_fast(
    t, r, th, p, alpha, v_r=None, v_th=None, v_p=None
):
    pos = [t, r, th, p, alpha] # alpha is not a coordinate
    vel = [v_r, v_th, v_p]
    if not any(vel): # if any of the velocity components are None
        return bl_to_cartesian_pos(*pos)
    return [*bl_to_cartesian_pos(*pos), *bl_to_cartesian_vel(*pos, *vel)]

# Converts positions
@jit
def bl_to_cartesian_pos(t, r, th, p, alpha):
    ...

# Converts velocities
@jit
def bl_to_cartesian_vel(t, r, th, p, alpha, v_r, v_th, v_p):
    ...
```
You can remove `_velocities_provided` everywhere after this.

In case you are interested in this refactor, please create a separate issue, so that, it can be properly tracked. Also, tag that issue in this PR.